### PR TITLE
Fix fallback client extend issue

### PR DESCRIPTION
**Pull request by blakecduncan**

### CODE REVIEW

File: packages/core/clients/fallbackKernelAccountClient.ts [hunk 1]
```diff
@@ -11,50 +11,79 @@ export const createFallbackKernelAccountClient = <
 >(
     clients: Array<
         KernelAccountClient<TEntryPoint, TTransport, TChain, TSmartAccount>
-    >
+    >,
+    onError?: (error: Error, clientUrl: string) => Promise<void>
 ): KernelAccountClient<TEntryPoint, TTransport, TChain, TSmartAccount> => {
+    // Map to store methods extended at runtime.
+    // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+    const extendedMethods = new Map<PropertyKey, any>()
```
Comment (blakecduncan) [inline packages/core/clients/fallbackKernelAccountClient.ts]: @adnpark just curious, is the extendMethods object needed?  I don't see us ever setting any values on it

File: packages/core/clients/fallbackKernelAccountClient.ts [hunk 2]
```diff
@@ -11,50 +11,79 @@ export const createFallbackKernelAccountClient = <
...
+                    for (const [key, modification] of Object.entries(
+                        modifications
+                    )) {
+                        if (typeof modification === "function") {
+                            // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                            ;(proxyClient as any)[key] = async (
+                                // biome-ignore lint/suspicious/noExplicitAny: <explanation>
+                                ...args: any[]
+                            ) => {
+                                return modification(...args)
```
Comment (blakecduncan) [inline packages/core/clients/fallbackKernelAccountClient.ts:61]: I think this should fix the extend issue.  There was an `await` here which was causing the extend function to wrap the return value in a promise

File: packages/core/clients/fallbackKernelAccountClient.ts (hunk 1 above)
Comment (adnpark) [reply packages/core/clients/fallbackKernelAccountClient.ts]: yeah it makes sense. right, we don't need it.

File: packages/core/clients/fallbackKernelAccountClient.ts (hunk 2 above)
Comment (adnpark) [reply packages/core/clients/fallbackKernelAccountClient.ts:61]: oh that makes sense. that's why it does not work after extending. great catch!

**Pull request merged**